androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java

```diff
@@ -19,59 +20,88 @@
 
+import android.app.Activity;
 import android.content.Context;
 import android.content.Intent;
+import android.content.pm.ActivityInfo;
+import android.os.Build;
 import android.os.Bundle;
 import android.os.RemoteException;
+import android.os.SystemClock;
 import android.support.test.InstrumentationRegistry;
 import android.support.test.rule.ActivityTestRule;
 import android.support.test.runner.AndroidJUnit4;
 import android.support.test.uiautomator.UiDevice;
 import android.test.suitebuilder.annotation.LargeTest;
 
-import static org.junit.Assert.assertTrue;
 import com.owncloud.android.R;
+import com.owncloud.android.lib.common.utils.Log_OC;
+import com.owncloud.android.utils.AccountsManager;
 
+import org.junit.After;
 import org.junit.Before;
+import org.junit.FixMethodOrder;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.junit.runners.MethodSorters;
 
 import java.lang.reflect.Field;
 
-import android.app.Activity;
-
 import static android.support.test.espresso.Espresso.onView;
 import static android.support.test.espresso.action.ViewActions.click;
 import static android.support.test.espresso.action.ViewActions.closeSoftKeyboard;
-import static android.support.test.espresso.action.ViewActions.typeText;
-
+import static android.support.test.espresso.action.ViewActions.replaceText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
 import static android.support.test.espresso.matcher.ViewMatchers.isEnabled;
 import static android.support.test.espresso.matcher.ViewMatchers.withId;
-
-import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
 import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertTrue;
+
 
 @RunWith(AndroidJUnit4.class)
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
 @LargeTest
 public class AuthenticatorActivityTest {
 
     public static final String EXTRA_ACTION = "ACTION";
     public static final String EXTRA_ACCOUNT = "ACCOUNT";
 
-    private static final int WAIT_LOGIN = 5000;
+    private static final int WAIT_INITIAL = 500;
+    private static final int WAIT_LOGIN = 4000;
+    private static final int WAIT_CONNECTION = 2000;
 
     private static final String ERROR_MESSAGE = "Activity not finished";
+    private static final String SUFFIX_BROWSER = "/index.php/apps/files/";
     private static final String RESULT_CODE = "mResultCode";
+    private static final String LOG_TAG = "LoginSuite";
+    private static final String USER_INEXISTENT = "userinexistent";
+
+    private Context targetContext = null;
 
+    private String testUser = null;
+    private String testUser2 = null;
+    private String testPassword = null;
+    private String testPassword2 = null;
+    private String testServerURL = null;
+    private int trusted = -1;
+
+    /*
```

Comment:
Yes, defining an enum type with those values an use the constants of the enum to make it easier to read all over the code.

Example:https://github.com/owncloud/android/blob/master/src/com/owncloud/android/datamodel/UploadsStorageManager.java#L49